ctapipe_io_lst/__init__.py

```diff
@@ -484,26 +492,32 @@ def fill_trigger_info(self, array_event, zfits_event):
         tib_available = lst.evt.extdevices_presence & 1
         ucts_available = lst.evt.extdevices_presence & 2
 
-        # Since at least June 2020, ucts should be more reliable
-        # before there were runs when ucts was missing or was shifted
-        # by one event after an event with ucts_trigger_type == 42.
-        # In case that happens we swtich to tib triggers
-        if tib_available and self.ucts_42_found:
+        # decide which source to use, if both are available,
+        # the option decides, if not, fallback to the avilable source
+        # if no source available, warn and do not fill trigger info
+        if tib_available and ucts_available:
+            if self.default_trigger_type == 'ucts':
+                trigger_bits = lst.evt.ucts_trigger_type
+            else:
+                trigger_bits = lst.evt.tib_masked_trigger
+
+        elif tib_available:
             trigger_bits = lst.evt.tib_masked_trigger
 
-        elif ucts_available and not self.ucts_42_found:
+        elif ucts_available:
             trigger_bits = lst.evt.ucts_trigger_type
-            if lst.evt.ucts_trigger_type == 42:
-                self.ucts_42_found = True
-                self.log.warning(
-                    'Event with UCTS trigger_type 42 found.'
-                    ' Probably means unreliable or shifted UCTS data.'
-                    ' Switching to TIB if available.'
-                )
+
         else:
             self.log.warning('No trigger info available.')
             return
 
+        if ucts_available and lst.evt.ucts_trigger_type == 42:
+            self.log.warning(
+                'Event with UCTS trigger_type 42 found.'
+                ' Probably means unreliable or shifted UCTS data.'
+                ' Switching to TIB if available.'
```

Comment:
Not switching now. I would not even suggest to switch, since for post-June 25th data some 42's appear, yet the UCTS event types seem ok with the implemented fix.

Reply:
right. I will add a comment about the option.